heron/healthm/src/java/com/twitter/heron/healthmgr/HealthManager.java

```diff
@@ -0,0 +1,355 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.healthmgr;
+
+import java.io.FileNotFoundException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.logging.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Singleton;
+import com.google.inject.name.Names;
+import com.microsoft.dhalion.api.IHealthPolicy;
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.events.EventManager;
+import com.microsoft.dhalion.policy.PoliciesExecutor;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.twitter.heron.healthmgr.common.HealthMgrConstants;
+import com.twitter.heron.healthmgr.common.PackingPlanProvider;
+import com.twitter.heron.healthmgr.sensors.TrackerMetricsProvider;
+import com.twitter.heron.scheduler.client.ISchedulerClient;
+import com.twitter.heron.scheduler.client.SchedulerClientFactory;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.ConfigLoader;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Key;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+/**
+ * e.g. options
+ * -d ~/.heron -p ~/.heron/conf/local -c local -e default -r userName -t AckingTopology
+ */
+public class HealthManager {
+  private static final Logger LOG = Logger.getLogger(HealthManager.class.getName());
+  private final Config config;
+  private final String trackerURL;
+
+  private Config runtime;
+  private Injector injector;
+  private SchedulerStateManagerAdaptor stateMgrAdaptor;
```

Comment:
The `HealthManager` will fetch topology info and scheduler location from the state manager.